src/interface/viewport.h

```diff
@@ -76,6 +76,43 @@ enum {
 	VIEWPORT_INTERACTION_MASK_BANNER = ~(1 << (VIEWPORT_INTERACTION_ITEM_BANNER - 2)), // Note the -2 for BANNER
 };
 
+typedef struct paint_struct paint_struct;
+
+struct paint_struct{
```

Comment:
It looks like there are two types of paint struct. The full `0x34` ps and the mini ps aka attached ps size `0x12`. We should probably break up the struct into these two and go through the code and make it obvious which is which. It's likely that it is used interchangeably in places.